src/compiler/program.ts

```diff
@@ -2986,10 +2986,6 @@ namespace ts {
                 }
             }
 
-            if (options.paths && options.baseUrl === undefined) {

```

Comment:
Do we need error when pathsBasePath  is not set instead (eg. apis using compilerOptions without passing it through the parseConfig just getting json value and passing it along?

Reply:
Hmm, yes, we do need to do something, but it would be best if we didn’t have to require `baseUrl` for that, since it allows resolving non-relative paths. We could use the host’s current directory? It looks like it should always be defined when coming from a CompilerHost. It’s optional on ModuleResolutionHost only when resolving non-relative `extends` values in the config parser.

Reply:
Changed moduleNameResolver to fall back to `host.getCurrentDirectory()` if `baseUrl` and `pathsBasePath` are both missing.